extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -419,21 +419,33 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 
 	async findAndExpandParentNode(notebookPath: string): Promise<BookTreeItem> {
 		let bookItem: BookTreeItem = this.currentBook?.getNotebook(notebookPath);
-		// if the node is not expanded getNotebook returns undefined, try to expand the parent node or getChildren of
-		// the root node.
-		if (!bookItem) {
-			// get the parent node and expand it if it's not already
-			let allNodes = this.currentBook?.getAllNotebooks();
-			let book = allNodes ? Array.from(allNodes?.keys())?.filter(x => x.indexOf(notebookPath.substring(0, notebookPath.lastIndexOf(path.sep))) > -1) : undefined;
-			let bookNode = book?.length > 0 ? this.currentBook?.getNotebook(book.find(x => x.substring(0, x.lastIndexOf(path.sep)) === notebookPath.substring(0, notebookPath.lastIndexOf(path.sep)))) : undefined;
-			if (bookNode) {
-				if (this._bookViewer?.visible) {
-					await this._bookViewer.reveal(bookNode, { select: true, focus: false, expand: 3 });
-				} else {
-					await this.getChildren(bookNode);
+		this.currentBook = this.books.length > 1 ? this.books.find(b => notebookPath.indexOf(b.bookPath) > -1) : this.currentBook;
```

Comment:
When would we ever expect to not be able to find the book when doing the find? That seems like an error - it would imply that the item we're trying to reveal is for a book that doesn't exist in the tree view as far as I understand it.

Reply:
when we open a notebook directly from open file which the viewlet isn't aware of, we end up not able to find the notebook in opened books.

Reply:
But why would we assume that it's in the "current" book (whatever that means)? Why not instead search all open books since it could be in any of them?

But we should also be able to just look at the root paths for each of the open books and see if it's a parent of the Notebook, right? I believe that all items in a book have to be under the root folder so that should be enough.

Can books be nested? If so that makes things much more difficult too...